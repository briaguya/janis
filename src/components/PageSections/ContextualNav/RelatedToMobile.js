import React from 'react';
import { injectIntl } from 'react-intl';
import { Link } from 'react-router-dom';
import { misc as i18n } from 'js/i18n/definitions';

const RelatedToMobile = ({ relatedTo, offeredBy, intl }) => (
  <div className="coa-RelatedToMobile">
    <div className="wrapper container-fluid">
      {!!relatedTo && !!relatedTo.length && (
        <div className="col-xs-12">
          <h2 className="coa-RelatedToMobile__title">
            {intl.formatMessage(i18n.relatedTo)}
          </h2>
          <ul>
            {relatedTo.map((relatedLinkData, index) => (
              <li key={index} className="coa-RelatedToMobile__item">
                <Link
                  to={`/${intl.locale}${relatedLinkData.url}`}
                  className="coa-RelatedToMobile__link"
                >
                  {relatedLinkData.title}
                </Link>
              </li>
            ))}
          </ul>
        </div>
      )}
      {!!offeredBy && !!offeredBy.length && (
        <div className="col-xs-12">
          <h2
            className={
              relatedTo.length && offeredBy.length
                ? 'coa-RelatedToMobile__second-title'
                : 'coa-RelatedToMobile__title'
            }
          >
            {intl.formatMessage(i18n.offeredBy)}
          </h2>
          <ul>
            {offeredBy.map((department, index) => (
              <li key={index} className="coa-RelatedToMobile__item">
                <Link
                  to={`/${intl.locale}${department.url}`}
                  className="coa-RelatedToMobile__link"
                >
                  {department.title}
                </Link>
              </li>
            ))}
          </ul>
        </div>
      )}
    </div>
  </div>
);

export default injectIntl(RelatedToMobile);
