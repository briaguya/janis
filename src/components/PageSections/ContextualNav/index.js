import React, { Fragment } from 'react';
import { injectIntl } from 'react-intl';

const ContextualNav = ({
  topic,
  topics,
  topiccollection,
  theme,
  department,
}) => {
  // Set the parent, if we have a topic, it's because we nav'd from a topic
  // if we don't, use the dept
  const parent = {
    slug: topic
      ? `/${theme.slug}/${topiccollection.slug}/${topic.slug}`
      : `/${department.slug}`,
    title: topic ? topic.title : department.title,
  };

  // Set the related links
  // TODO
  const related = [
    {
      slug: `#`,
      title: `Placeholder`,
    },
    {
      slug: `#`,
      title: `Placeholder`,
    },
  ];

  return (
    <div className="coa-ContextualNav">
      <div className="wrapper container-fluid">
        <div className="coa-ContextualNav__container">
          <div className="coa-ContextualNav__parent">
            <a href={parent.slug} className="coa-ContextualNav__arrow">
              <i className="material-icons">arrow_back</i>
              <span>{parent.title}</span>
            </a>
          </div>
          <div className="coa-ContextualNav__related">
            <span className="coa-ContextualNav__label">Related to: </span>
            {related.map((topic, index) => (
              <a key={index} href={topic.slug}>
                {topic.title}
                {index !== related.length - 1 && ', '}
              </a>
            ))}
          </div>
          <div className="coa-ContextualNav__dept">
            {department && (
              <Fragment>
                <span className="coa-ContextualNav__label">Offered by: </span>
                <a href={`/${department.slug}`}>{department.title}</a>
              </Fragment>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
export default injectIntl(ContextualNav);
