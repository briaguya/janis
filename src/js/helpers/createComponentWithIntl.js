// this is a helper function for the jest snapshot tests
// https://github.com/formatjs/react-intl/blob/master/docs/Testing-with-React-Intl.md#snapshot-testing

import React from 'react';
import renderer from 'react-test-renderer';
import {IntlProvider} from 'react-intl';

const createComponentWithIntl = (children, props = {locale: 'en'}) => {
  return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};

export default createComponentWithIntl;
