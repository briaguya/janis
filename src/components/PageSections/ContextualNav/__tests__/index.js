import React from 'react';
import renderer from 'react-test-renderer';
// BrowserRouter is necessary so the tests know where to find Link
import { BrowserRouter } from 'react-router-dom';

import createComponentWithIntl from 'js/helpers/createComponentWithIntl';
import ContextualNav from '../../ContextualNav'
import RelatedToMobile from '../../ContextualNav/RelatedToMobile/'

describe('ContextualNav snapshot', () => {
  test('ContextualNav renders', () => {
    const parent = {
      id: "VG9waWNOb2RlOjUz",
      title: "Disease prevention",
      url: "/health-safety/healthcare-prevention/disease-prevention/"
    };
    const relatedTo = [
      {
        id: "VG9waWNOb2RlOjU3",
        title: "Household waste",
        url: "/housing-utilities/recycling-trash-and-compost/household-waste/"
      },
      {
        id: "VG9waWNOb2RlOjU5",
        title: "Recycling and reuse",
        url: "/housing-utilities/recycling-trash-and-compost/recycling-reuse/"
      }
    ];
    const offeredBy = [
      {
        id: "RGVwYXJ0bWVudFBhZ2VOb2RlOjExNw==",
        title: "Austin Public Health",
        url: "/austin-public-health/"
      }
    ];

    const component = createComponentWithIntl(
      <BrowserRouter>
      	<ContextualNav
          parent={parent}
          relatedTo={relatedTo}
          offeredBy={offeredBy}
      	/>
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('RelatedToMobile snapshot', () => {
  test('RelatedToMobile renders', () => {
    const relatedTo = [
      {
        id: "VG9waWNOb2RlOjU3",
        title: "Household waste",
        url: "/housing-utilities/recycling-trash-and-compost/household-waste/"
      },
      {
        id: "VG9waWNOb2RlOjU5",
        title: "Recycling and reuse",
        url: "/housing-utilities/recycling-trash-and-compost/recycling-reuse/"
      }
    ];
    const offeredBy = [
      {
        id: "RGVwYXJ0bWVudFBhZ2VOb2RlOjExNw==",
        title: "Austin Public Health",
        url: "/austin-public-health/"
      }
    ];

    const component = createComponentWithIntl(
      <BrowserRouter>
        <RelatedToMobile
          relatedTo={relatedTo}
          offeredBy={offeredBy}
        />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

