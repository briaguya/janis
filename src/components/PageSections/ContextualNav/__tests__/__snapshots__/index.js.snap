// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualNav snapshot ContextualNav renders 1`] = `
<div
  className="coa-ContextualNav"
>
  <div
    className="wrapper container-fluid"
  >
    <div
      className="coa-ContextualNav__container"
    >
      <div
        className="coa-ContextualNav__parent"
      >
        <a
          className="coa-ContextualNav__arrow"
          href="/en/health-safety/healthcare-prevention/disease-prevention/"
          onClick={[Function]}
        >
          <span>
            Disease prevention
          </span>
        </a>
      </div>
      <div
        className="coa-ContextualNav__related"
      >
        <span
          className="coa-ContextualNav__label"
        >
          Related to: 
        </span>
        <a
          href="/en/housing-utilities/recycling-trash-and-compost/household-waste/"
          onClick={[Function]}
        >
          Household waste
          , 
        </a>
        <a
          href="/en/housing-utilities/recycling-trash-and-compost/recycling-reuse/"
          onClick={[Function]}
        >
          Recycling and reuse
        </a>
      </div>
      <div
        className="coa-ContextualNav__dept"
      >
        <span
          className="coa-ContextualNav__label"
        >
          Offered by: 
        </span>
        <a
          href="/en/austin-public-health/"
          onClick={[Function]}
        >
          Austin Public Health
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`RelatedToMobile snapshot RelatedToMobile renders 1`] = `
<div
  className="coa-RelatedToMobile"
>
  <div
    className="wrapper container-fluid"
  >
    <div
      className="col-xs-12"
    >
      <h2
        className="coa-RelatedToMobile__title"
      >
        Related to
      </h2>
      <ul>
        <li
          className="coa-RelatedToMobile__item"
        >
          <a
            className="coa-RelatedToMobile__link"
            href="/en/housing-utilities/recycling-trash-and-compost/household-waste/"
            onClick={[Function]}
          >
            Household waste
          </a>
        </li>
        <li
          className="coa-RelatedToMobile__item"
        >
          <a
            className="coa-RelatedToMobile__link"
            href="/en/housing-utilities/recycling-trash-and-compost/recycling-reuse/"
            onClick={[Function]}
          >
            Recycling and reuse
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col-xs-12"
    >
      <h2
        className="coa-RelatedToMobile__second-title"
      >
        Offered by
      </h2>
      <ul>
        <li
          className="coa-RelatedToMobile__item"
        >
          <a
            className="coa-RelatedToMobile__link"
            href="/en/austin-public-health/"
            onClick={[Function]}
          >
            Austin Public Health
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
